src/sentry/event_manager.py

```diff
@@ -2418,19 +2418,37 @@ def _save_aggregate_performance(jobs: Sequence[PerformanceJob], projects: Projec
 
 @metrics.wraps("performance.performance_issue.should_create_group", sample_rate=1.0)
 def should_create_group(client: Any, grouphash: str, type: GroupType) -> bool:
-    times_seen = client.incr(f"grouphash:{grouphash}")
-    metrics.incr(
-        "performance.performance_issue.grouphash_counted",
-        tags={"times_seen": times_seen, "group_type": GROUP_TYPE_TO_TEXT.get(type, "Unknown Type")},
-        sample_rate=1.0,
-    )
+    with sentry_sdk.start_span(op="event_manager.should_create_group") as span:
+        times_seen = client.incr(f"grouphash:{grouphash}")
+        metrics.incr(
+            "performance.performance_issue.grouphash_counted",
+            tags={
+                "times_seen": times_seen,
+                "group_type": GROUP_TYPE_TO_TEXT.get(type, "Unknown Type"),
+            },
+            sample_rate=1.0,
+        )
 
-    if times_seen >= GROUPHASH_IGNORE_LIMIT_MAP.get(type, DEFAULT_GROUPHASH_IGNORE_LIMIT):
-        client.delete(grouphash)
-        return True
-    else:
-        client.expire(grouphash, 60 * 60 * 24)  # 24 hour expiration from last seen
-        return False
+        if times_seen >= GROUPHASH_IGNORE_LIMIT_MAP.get(type, DEFAULT_GROUPHASH_IGNORE_LIMIT):
+            client.delete(grouphash)
+            metrics.incr(
+                "performance.performance_issue.issue_will_be_created",
+                tags={"group_type": GROUP_TYPE_TO_TEXT.get(type, "Unknown Type")},
+                sample_rate=1.0,
+            )
+
+            # TODO: Experimental tag to indicate when a Slow DB Issue would have been created.
+            # This is in place to confirm whether the above rate limit is effective in preventing spikes
+            # and inaccuracies.
+            if type == GroupType.PERFORMANCE_SLOW_SPAN:
```

Comment:
I think you might need to return false from this condition because if `type == GroupType.PERFORMANCE_SLOW_SPAN` is true but `if span.containing_transaction:` is false, you end up returning true

Reply:
oh wait the return on line 2446 is for the `if type == GroupType.PERFORMANCE_SLOW_SPAN:` right?

Reply:
Oh crap 😱 

I didn't realize I accidentally pushed this. I returned True here because I was checking something with the test. Thanks for catching this!

Reply:
Ok this gave me a fright for a second because I was actually returning True at one point to confirm my test was failing 🤣 

But yeah, the `if span.containing_transaction:` block is specifically for setting the transaction tag, this will still return False regardless if it is a slow span issue

Reply:
ok yeah this should be fine. the linebreak on my screen confused me lol